docs/compatibility.md

```diff
@@ -1,3 +1,76 @@
+### v1.3.18
+
+Some ops have different implementations on different devices. Lots of macros and type checks are performed which make the codes hard to maintain. For example:
```

Comment:
Lots of macros and type checks are scattered in several files, which makes the code hard to maintain.